src/main/java/org/elasticsearch/percolator/PercolateContext.java

```diff
@@ -693,4 +690,79 @@ public InnerHitsContext innerHits() {
         throw new UnsupportedOperationException();
     }
 
+    // TODO needs real implementation?
```

Comment:
I don't think so. The percolator isn't executing requests during query parsing.
Maybe add `assert false : "percolator should get here..."? So if in the future this does happen to be the case the test will fail and then we know about it